stdlib/ballerina-builtin/src/main/ballerina/ballerina/security/authentication/basic/authn-interceptor.bal

```diff
@@ -0,0 +1,65 @@
+package ballerina.security.authentication.basic;
+
+import ballerina.net.http;
+import ballerina.log;
+import ballerina.security.utils;
+import ballerina.security.crypto;
+
+@Description {value:"Authentication cache name"}
+const string AUTH_CACHE = "basic_auth_cache";
+
+@Description {value:"Basic authenticator instance"}
+BasicAuthenticator authenticator;
+
+@Description {value:"Intercepts a request for authentication"}
+@Param {value:"req: InRequest object"}
+@Return {value:"boolean: true if authentication is a success, else false"}
+public function handle (http:InRequest req) (boolean) {
+    // extract the header value
+    string basicAuthHeaderValue;
+    error err;
+    basicAuthHeaderValue, err = utils:extractBasicAuthHeaderValue(req);
```

Comment:
Cant we use `var basicAuthHeaderValue, err = utils:extractBasicAuthHeaderValue(req);` here?